reactor-netty-http/src/main/java/reactor/netty/http/client/Http2Pool.java

```diff
@@ -80,6 +80,10 @@
  *     <li>{@link PoolMetrics#idleSize()} always returns {@code 0}.</li>
  * </ul>
  * <p>
+ * If minimum connections is specified, the cached connections with active streams will be kept at that minimum
+ * (can be the best effort). However, if the cached connections have reached max concurrent streams,
+ * then new connections will be allocated up to the maximum connections limit.
```

Comment:
Hmm, so the code now will do this, right:
For any new HTTP/2 request, if the number of socket connections is below the minimum number of connections, a new socket connection will get created and the HTTP/2 stream will get created from that channel, right?

I don't really get that from the javadoc, but that may just be me.

What does the best effort part mean?  I think I'm missing that part in the code and want to make sure I'm not misunderstanding something?

Reply:
> Hmm, so the code now will do this, right: For any new HTTP/2 request, if the number of socket connections is below the minimum number of connections, a new socket connection will get created and the HTTP/2 stream will get created on that new socket connection, right?

yes

> I don't really get that from the javadoc, but that may just be me.
> 
> What does the best effort part mean? I think I'm missing that part in the code and want to make sure I'm not misunderstanding something?

As we do not have a strong synchronisation between when we return the connection and we acquire the connection we may not be able to stay exactly on the min number that is specified.